Assignment 2.Rmd

```diff
@@ -81,20 +101,137 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
-````
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+# first, lets add the valiable n_actions (number of disciplinary actions) to our graph database, to count and save the number of disciplinary actions that each student received and each teacher initiated
+std_actions <- count(EDGE,to)
+tch_actions <- count(EDGE,from)
+
+#make sure variables have proper (mathing) names 
+names(std_actions) <- c("id","n_actions")
+names(tch_actions) <- c("id","n_actions")
 
+#add new variable to student and teacher vectors
+V.STD <- merge(V.STD,std_actions, by = "id")
+V.TCH <- merge(V.TCH,tch_actions, by = "id")
+
+#update VERTEX data with our new variable "n_actions"
+VERTEX <- bind_rows(V.TCH, V.STD)
+
+#update graph data with our new variable "n_actions"
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+```
+
+
+Now, let's plot!
+```{r}
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count, vertex.size=3*VERTEX$n_actions+10)
+
+```
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+Let's clean up the workspace, get the new data and do some cleaning...
+```{r}
+#clean up and get data
+rm(list = ls())
+D1 <- read.csv("~/R/HUDK/assignment2/hudk4050-classes.csv")
+
+#rename rows with student names and drop student names
+rownames(D1) <- paste(D1$First.Name,D1$Last.Name)
+D1 <- D1[,3:8] 
+
+#standardize space/no space in how subjects are refered to and make it a string
+D1[,1] <- gsub(" ", "", as.character(D1[,1]), fixed = TRUE)
+D1[,2] <- gsub(" ", "", as.character(D1[,2]), fixed = TRUE)
+D1[,3] <- gsub(" ", "", as.character(D1[,3]), fixed = TRUE)
+D1[,4] <- gsub(" ", "", as.character(D1[,4]), fixed = TRUE)
+D1[,5] <- gsub(" ", "", as.character(D1[,5]), fixed = TRUE)
+D1[,6] <- gsub(" ", "", as.character(D1[,6]), fixed = TRUE)
+
```

Comment:
This is fine. But could be redundant if there are more columns. Try "lapply" to get this done with one line.